front/src/app/scenes/transaction/transaction.component.ts

```diff
@@ -2,22 +2,26 @@
 import {Component, OnDestroy, OnInit} from '@angular/core';
 import {ActivatedRoute, ParamMap} from '@angular/router';
 import {forkJoin, interval, Observable, of, Subscription} from 'rxjs';
-import {mergeMap, startWith, tap} from 'rxjs/operators';
+import {catchError, map, mergeMap, startWith, tap} from 'rxjs/operators';
 import {fromPromise} from 'rxjs/internal-compatibility';
 /*SERVICES*/
 import {CommonService} from '../../services/common.service';
 import {LayoutService} from '../../services/layout.service';
 import {WalletService} from '../../services/wallet.service';
 import {MetaService} from '../../services/meta.service';
 /*MODELS*/
-import {ProcessedLog, ProcessedLogItem, Transaction, TxLog} from '../../models/transaction.model';
-import {ContractEventsAbi} from '../../utils/types';
+import {ProcessedLog, ProcessedABIItem, Transaction, TxLog, ProcessedABIData} from '../../models/transaction.model';
+import {ContractAbiByID, ContractEventsAbi} from '../../utils/types';
 import Web3 from 'web3';
-import {AbiItem} from 'web3-utils';
+import {AbiItem, AbiInput} from 'web3-utils';
 import {Address} from '../../models/address.model';
+import {Contract} from '../../models/contract.model';
 /*UTILS*/
 import {AutoUnsubscribe} from '../../decorators/auto-unsubscribe';
 import {META_TITLES} from '../../utils/constants';
+import {ErcName} from "../../utils/enums";
+
+import CID from "../../../../node_modules/cids/dist/index.min.js"
```

Comment:
Any suggestions for a better way to do this?

Reply:
@jmank88 const CID = require('cids') from docs, i've checked there's no typing for cids